### PR TITLE
Vcb 6 leds on ct board

### PR DESCRIPTION
**Pull request by akrosis**

### CODE REVIEW

File: src/components/board/led/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  public render(): React.ReactNode {
+    return (
+      <div className="led-container">
+        {Object.keys(this.state)
+          .map(Number)
+          .sort((n1, n2) => n2 - n1)
+          .map((key) => (
+            <div key={'led-' + key}>
+              <div className="label">{'S' + key}</div>
```
Comment (stefenz) [inline src/components/board/led/index.tsx:46]: The LEDs should not be labeled with S*. Please label it like on the real board.

**Review by stefenz — CHANGES_REQUESTED** [review]
Looks good to me, just the labelling in the UI is not correct yet

File: src/components/board/led/index.tsx (hunk 1 above)
Comment (stefenz) [reply src/components/board/led/index.tsx:46]: And please merge in the actual master so this pr is ready to merge

Comment (stefenz) [reply src/components/board/led/index.tsx:46]: And add some tests ;)

Comment (stefenz) [reply src/components/board/led/index.tsx]: Sorry about the tests - my fault.

Unfortunately the labels are now very tiny and not readable. Wouldn't you label them like on the physical board, where just the leds at the edges of each four are  labeled with numbers and between with dots (...)?

**Review by m-dot-wigger — CHANGES_REQUESTED** [review]
install npm package to get imports in alphabetical order and format files

as mentioned of Stefan only label highest and lowest so that labels can be bigger.

File: src/board/devices/output/leds.ts [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
-import { Word } from 'types/binary'
+import { Byte, Word } from 'types/binary'
 
 export class LEDs extends Device {
-  public startAddress = Word.fromUnsignedInteger(0x08000000)
-  public endAddress = Word.fromUnsignedInteger(0x081fffff)
+  public startAddress = Word.fromUnsignedInteger(0x60000100)
+  public endAddress = Word.fromUnsignedInteger(0x60000103)
   public isReadOnly = false
   public isVolatile = false
```
Comment (clooudch) [inline src/board/devices/output/leds.ts:8]: Should probably be `true`. I think LED's lose their state when the board is reset.

File: src/components/board/led/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+type LedState = {
+  [key: number]: boolean
+}
+
+type LedProps = {
+  startIndex: number
+  size: number
+}
+
+export class Led extends React.Component<LedProps, LedState> {
```
[the remark this answers is not in the document]
Comment (akrosis) [reply src/components/board/led/index.tsx:14]: Sounds good to me, should we do this in a seperate branch, after everything is merged from this sprint?
Or should I do the changes in this branch?

File: src/components/board/index.tsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React from 'react'
 import { DipSwitch } from './dip-switch'
 
 import './style.css'
+import { Led } from './led'
 
 export class Board extends React.Component {
```
[the remark this answers is not in the document]
Comment (akrosis) [reply src/components/board/index.tsx:7]: Same question as with te suffixes, should I do this here or do we make a seperate branch after everything from the other stories is merged?

File: src/components/board/index.tsx [hunk 5]
```diff
@@ -2,6 +2,7 @@ import React from 'react'
 import { DipSwitch } from './dip-switch'
 
 import './style.css'
+import { Led } from './led'
```
[the remark this answers is not in the document]
Comment (akrosis) [reply src/components/board/index.tsx:4]: Im not sure I understand what you mean here, the order of the imports is alphabetical right?

[the remark this answers is not in the document]
Comment (clooudch) [reply src/components/board/index.tsx:4]: I am sorry. I have created this comment some time ago before we had this new import sort plugin for prettier.

File: src/components/board/led/index.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  private readonly endIndex: number
+
+  constructor(props: LedProps) {
+    super(props)
+    this.endIndex = this.props.startIndex + this.props.size - 1
+    Board.processor.on('afterCycle', this.update())
+    this.state = this.getState()
+  }
+
+  private update() {
```
[the remark this answers is not in the document]
Comment (clooudch) [reply src/components/board/led/index.tsx:24]: Yes. Perfect. Thank you 😄 .

File: src/components/board/led/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (clooudch) [reply src/components/board/led/index.tsx:14]: For the files which are affected by you PR you can change it here and for the other files we change it along the way. I don't think that we need a dedicated User Story or Bug for this. It's actually only a very small change.

File: src/components/board/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (clooudch) [reply src/components/board/index.tsx:7]: I close this comment here. As it's covered in another conversation.

File: src/components/board/led/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (akrosis) [reply src/components/board/led/index.tsx]: done

**Review by clooudch — APPROVED** [review]

**Review by m-dot-wigger — APPROVED** [review]

**Pull request merged**